File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        string st, ans = "";
        
        stringstream ss(s);
        
        while(ss>>st){
            
            if(ans != "") ans = ' ' + ans;
            
            ans = st + ans;
        }
        return ans;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] array = s.trim().split("\\s+");
        StringBuilder res = new StringBuilder();
        
        for(int i=array.length -1; i>=0; i--){
            res.append(array[i]);
            
            if(i!=0) res.append(" ");
        }
        return res.toString();
    }
}